src/Mod/Sketcher/Gui/Utils.cpp

```diff
@@ -332,15 +365,37 @@ double SketcherGui::GetPointAngle(const Base::Vector2d& p1, const Base::Vector2d
 
 // Set the two points on circles at minimal distance
 // in concentric case set points on relative X axis
-void SketcherGui::GetCirclesMinimalDistance(const Part::GeomCircle* circle1,
-                                            const Part::GeomCircle* circle2, Base::Vector3d& point1,
+void SketcherGui::GetCirclesMinimalDistance(const Part::Geometry* geom1,
+                                            const Part::Geometry* geom2,
+                                            Base::Vector3d& point1,
                                             Base::Vector3d& point2)
 {
-    double radius1 = circle1->getRadius();
-    double radius2 = circle2->getRadius();
+    double radius1 = 0.0;
+    double radius2 = 0.0;
+
+    if (geom1->getTypeId() == Part::GeomCircle::getClassTypeId()) {
```

Comment:
General comment that don't call for action here : 
In sketcher we do a lot of this kind of 'if' to get geos points / radius / ... and the syntax is pretty heavy.
Perhaps we could create a Part::Geometry::getRadius that would return -1 if the geo is not a circle or an arc. This way we would remove these if(circle), else-if(arc), else-if.. that are all over the place.